GraphExplorerPermissionsService/Services/PermissionsStore.cs

```diff
@@ -115,48 +120,94 @@ private void SeedPermissionsTables()
 				}
 			});
             return scopesInformationDictionary;
         }
 
+        /// <summary>
+        /// Gets the permissions descriptions and their localized instances from DevX Content Repo.
+        /// </summary>
+        /// <param name="locale">The locale of the permissions descriptions file.</param>
+        /// <param name="org">The org or owner of the repo.</param>
+        /// <param name="branchName">The name of the branch with the file version.</param>
+        /// <returns>The localized instance of permissions descriptions.</returns>
+        private async Task<IDictionary<string, IDictionary<string, ScopeInformation>>> GetPermissionsFromGithub(string locale,
+                                                                                                                string org,
+                                                                                                                string branchName)
+        {
+            string host = _configuration["BlobStorage:GithubHost"];
+            string repo = _configuration["BlobStorage:RepoName"];
+
+            string localizedFilePathSource = FileServiceHelper.GetLocalizedFilePathSource(_permissionsContainerName, _scopesInformation, locale);
+
+            // Get the absolute url from configuration and query param, then read from the file
+            var queriesFilePathSource = string.Concat(host, org, repo, branchName, FileServiceConstants.DirectorySeparator, localizedFilePathSource);
+
+            // Construct the http request message
+            var httpRequestMessage = new HttpRequestMessage(HttpMethod.Get, queriesFilePathSource);
+
+            // Get file contents from source
+            string scopesInfoJson = await _httpClientUtility.ReadFromFile(httpRequestMessage);
+
+            var scopesInformationDictionary = await CreateScopesInformationTables(scopesInfoJson);
+
+            return scopesInformationDictionary;
+        }
+
+        /// <summary>
+        /// Creates a dictionary of scopes information
+        /// </summary>
+        /// <param name="filePath">The path of the file from Github.</param>
+        /// <param name="cacheEntry">An optional cache entry param.</param>
+        /// <returns>A dictionary of scopes information.</returns>
+        private async Task<IDictionary<string, IDictionary<string, ScopeInformation>>> CreateScopesInformationTables(string scopesInfoJson, ICacheEntry cacheEntry = null)
+        {
+            var _delegatedScopesInfoTable = new Dictionary<string, ScopeInformation>();
+            var _applicationScopesInfoTable = new Dictionary<string, ScopeInformation>();
+
+            if (string.IsNullOrEmpty(scopesInfoJson))
+            {
+                return null;
+            }
+
+            ScopesInformationList scopesInformationList = JsonConvert.DeserializeObject<ScopesInformationList>(scopesInfoJson);
+
+            foreach (ScopeInformation delegatedScopeInfo in scopesInformationList.DelegatedScopesList)
```

Comment:
```suggestion
           var _delegatedScopesInfoTable = new Dictionary<string, ScopeInformation>(scopesInformationList.DelegatedScopesList.Select(x => new KeyValuePair<string, ScopeInformation>(x.ScopeName, x));
```

This way you avoid:
- having to periodically grow the dictionary
- using a loop when the intend is to project data
- skipping on vectorialization